crates/hyperdrive-math/src/long/close.rs

```diff
@@ -115,4 +172,56 @@ mod tests {
         assert!(result.is_err());
         Ok(())
     }
+
+    // Tests market valuation against yield space valuation when closing a long
+    // with the minimum transaction amount.
+    #[tokio::test]
+    async fn test_calculate_market_value_long() -> Result<()> {
+        let tolerance = int256!(1e14);
+
+        // Fuzz the spot valuation and yield space valuation against each other.
+        let mut rng = thread_rng();
+        for _ in 0..*FAST_FUZZ_RUNS {
+            let state = rng
+                .gen::<State>()
+                .calculate_pool_state_after_add_liquidity(fixed!(1e27), true)?;
+            let bond_amount = state.minimum_transaction_amount();
+            let maturity_time = state.position_duration();
+            let current_time = rng.gen_range(fixed!(0)..=maturity_time);
+
+            // Ensure curve_fee is smaller than spot_price to avoid overflows
+            // on the yield space valuation, as that'd mean having to pay a larger
+            // amount of fees than the current value of the long
+            if state.curve_fee() > state.calculate_spot_price()? {
+                continue;
+            }
+
+            let yield_space_valuation = state.calculate_close_long(
```

Comment:
I think calling this `yield_space_valuation` is misleading, because it includes fees and flat + curve. It's really the hyperdrive valuation. Same in the short test.